libraries/lib-time-and-pitch/StaffPad/TimeAndPitch.cpp

```diff
@@ -498,6 +496,8 @@ void TimeAndPitch::feedAudio(const float* const* input_smp, int numSamples)
       if (_analysis_hop_counter >= hop_a)
       {
          _analysis_hop_counter -= hop_a;
+         d->hop_s_err += d->exact_hop_s - hop_s;
+         d->hop_a_err += d->exact_hop_a - hop_a;
```

Comment:
I don't feel competent to review this without much more time for study of the library.  I'll just ask:
- does this affect performance only or does it change the calculations?
- The reassignment of `hop_s_err` and `hop_a_err` used to happen once per function call.  Do you mean to make it happen at most once, or sometimes multiple times?
- If not multiple times, then should `hop_s` and `hop_a` be changed after these two assignments?
-

Reply:
The naming of these variables isn't really descriptive. As I understand this code, `hop_a_err` is the analysis samples "credit". `exact_hop_a`, the "exact analysis hop size", can be a fractional number of samples, but of course, one can only feed discrete samples at a time. So when you need 2.5 samples to perform one FFT, as soon as you get the 3rd sample, all 3 samples are processed, but you have a credit of 0.5 samples. This goes into `hop_a_err`. Same principle for `hop_s_err` but in the IFFT direction.
There were situations (when `numSamples` wasn't the 1024 "default" or the stretch ratio is so low that `exact_hop_s < 1`) where those state variables were updated although no analysis or no synthesis happened. As a result, no samples were ever produced, and we were falling in an endless loop.

Now to your questions:
* no, negligibly
* sometimes multiple times, which is the correct thing to do: update your credit variables if and only if analysis/processing happened

@dozzzzer also tested the other extrem: stretching from 30s to 12h. I also did and these changes do not help. Looking into it.